File: src/main/java/luminia/backend/models/TaskResult.java
package luminia.backend.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity(name = "TaskResult")
@Table(name = "task_results")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class TaskResult {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "task_results_gen")
    @SequenceGenerator(name = "task_results_gen", sequenceName = "task_results_seq", allocationSize = 1)
    @Column(name = "id", nullable = false)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "task_assign_id")
    private TaskAssign taskAssign;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "target_id")
    private User target;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id")
    private User author;
    private TaskStatus status;
    private int rating;
    private List<Long> attachments;
    @Column(name = "upload_date")
    private LocalDateTime uploadDate;

    public enum TaskStatus {
        REJECTED, STARTED, NOT_STARTED, WAIT, ACCEPT, ABSORBED
    }
}

File: frontend/src/services/remote/taskresult.ts
import type Attachment from "./attachment"
import type List from "./list"
import type TaskAssign from "./taskassign"
import type User from "./user"



export default class TaskResult {
    public id: number
    public taskAssign: TaskAssign
    public target: User
    public author: User
    public status: string
    public rating: number
    public attachments: List<Attachment>
    public uploadDate: string

    constructor(
        id: number,
        taskAssign: TaskAssign,
        target: User,
        author: User,
        status: string,
        rating: number,
        attachments: List<Attachment>,
        uploadDate: string
    ) {
        this.id = id
        this.taskAssign = taskAssign
        this.target = target
        this.author = author
        this.status = status
        this.rating = rating
        this.attachments = attachments
        this.uploadDate = uploadDate
    }
}